src/main/java/org/codeforamerica/messaging/config/SecurityConfiguration.java

```diff
@@ -28,4 +45,40 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .httpBasic(withDefaults());
         return http.build();
     }
+
+    private AuthorizationDecision validIP(Supplier<Authentication> authentication, RequestAuthorizationContext context) {
+            HttpServletRequest request = context.getRequest();
+            List<String> requestAddresses = getRequestAddresses(request);
+            boolean ipAddressAllowed = Arrays.stream(allowedIpAddresses.split(","))
+                    .anyMatch(allowedIpAddress -> {
+                        IpAddressMatcher ipAddressMatcher = new IpAddressMatcher(allowedIpAddress);
+                        return requestAddresses.stream().anyMatch(ipAddressMatcher::matches);
+                    });
+            return new AuthorizationDecision(ipAddressAllowed);
+    }
+
+    private List<String> getRequestAddresses(HttpServletRequest request) {
+        List<String> requestAddresses = new ArrayList<>(List.of(request.getRemoteAddr()));
```

Comment:
Do you need `new ArrayList<>`? I would have expected `List.of` to return a `List<E>` by itself, but I haven't run this code

Reply:
It is needed to make it a mutable list.